composables/zksync/useWithdrawalFinalization.ts

```diff
@@ -122,6 +122,13 @@ export default (transactionInfo: ComputedRef<TransactionInfo>) => {
       }
       const wallet = await onboardStore.getWallet();
       const { transactionParams, gasLimit, gasPrice } = (await estimateFee())!;
+
+      // Check if the batch has been processed before attempting finalization
+      const l1BatchNumber = finalizeWithdrawalParams.value?.l1BatchNumber;
+      if (l1BatchNumber && !(await isBatchProcessed(l1BatchNumber))) {
+        throw new Error("Batch not yet processed. Please wait and try again later.");
+      }
```

Comment:
This does not seem necessary as it doesnt seem to be the root of the issue. `Wallet.isWithdrawalFinalized()` is used already and provides that check for withdrawal finalization